Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -86,36 +213,30 @@ namespace AzAssetBrowserRequestHandlerPrivate
             return false;
         }
 
-        if (optionalProductAssetTypes.has_value())
```

Comment:
Note from nick:  This array was never filled in, in any code paths Ic ould find.  This means that the HasConflictingProducts API was never functional.